kai/kai_trace.py

```diff
@@ -37,15 +36,14 @@ class Trace:
 
     def __init__(
         self,
-        config: KaiConfig,
```

Comment:
I think tracing might be broken, it used the config for 2 things

1) Check if tracing is enabled/disabled
2) Get the log directory

I assume tracing will get up to the spot below and look for self.config.log_dir and not find it.

https://github.com/konveyor/kai/blob/9bfb7be7d78350cd18ea93001098bb3b6e8d43de/kai/kai_trace.py#L55

Reply:
I'm seeing the below when I run

[2024-07-23 17:32:55 -0400] [87706] [ERROR] Error handling request
Traceback (most recent call last):
  File "/Users/jmatthews/git/jwmatthews/kai/env/lib/python3.12/site-packages/aiohttp/web_protocol.py", line 452, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jmatthews/git/jwmatthews/kai/env/lib/python3.12/site-packages/aiohttp/web_app.py", line 543, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jmatthews/git/jwmatthews/kai/kai/routes/get_incident_solutions_for_file.py", line 44, in post_get_incident_solutions_for_file
    trace = KaiTrace(
            ^^^^^^^^^
  File "/Users/jmatthews/git/jwmatthews/kai/kai/kai_trace.py", line 55, in __init__
    log_dir = process_log_dir_replacements(self.config.log_dir)
                                           ^^^^^^^^^^^
AttributeError: 'KaiTrace' object has no attribute 'config'

Reply:
Let's see if my most recent commit fixes this

Reply:
I see logs written to disk and trace data written to disk.

We may need to tweak the Trace data.  Originally I had assumed that we'd create the Trace object once per request, currently we are creating it twice.  This functions, but the output is a little off where some of the layout is slightly off in regard to directory structure.  

This is a small tweak, the larger issue right now is that I can't run run_demo.py, it gets back a 200 response and has data that looks good, but the run_demo.py isn't able to parse the response to write it out.

Not sure what is wrong yet, almost looks like its not decoding to json and treating the response as a regular string

Reply:
A `Response`s `.json()` method returns a `str` not a `dict`. Wondering why this wasn't an issue for me before... Pushed a fix, let's see if that helps.